src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_kds.c

```diff
@@ -2304,6 +2304,51 @@ xocl_kds_xgq_cfg_scus(struct xocl_dev *xdev, xuid_t *xclbin_id, struct xrt_cu_in
 	return ret;
 }
 
+static int xocl_kds_xgq_cleanup_cus(struct xocl_dev *xdev)
+{
+	struct xgq_com_queue_entry resp = {};
+	struct xgq_cmd_uncfg_cu *uncfg_cu = NULL;
+	struct kds_sched *kds = &XDEV(xdev)->kds;
+	struct kds_client *client = NULL;
+	struct kds_command *xcmd = NULL;
+	int ret = 0;
+
+	client = kds->anon_client;
+	xcmd = kds_alloc_command(client, sizeof(struct xgq_cmd_uncfg_cu));
+	if (!xcmd)
+		return -ENOMEM;
+
+	uncfg_cu = xcmd->info;
+
+	uncfg_cu->hdr.opcode = XGQ_CMD_OP_CLEANUP_ALL_CU;
+	uncfg_cu->hdr.count = sizeof(*uncfg_cu) - sizeof(uncfg_cu->hdr);
```

Comment:
It looks weird that the opcode and the data struct name doesn't match.
Can we either add a new struct for XGQ_CMD_OP_CLEANUP_ALL_CU or instead of add a new opcode, use 1 reserved bit in xgq_cmd_uncfg_cu to let ERT know unconfigure all of the CUs if existed.

Personally, I will prefer to use use the reserved bit in uncfg_cu command. So that if the bit is set, ERT will unconfigure anything.

Reply:
I have updated based on your suggestion.  Thanks